python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -77,7 +78,10 @@ def metadata(self) -> ArbitraryMetadataMapping:
 
     @property
     def tags(self) -> Mapping[str, str]:
-        return self._spec.tags
+        return {
+            **self._spec.tags,
+            **{f"{KIND_PREFIX}{kind}": "true" for kind in self._spec.kinds or []},
```

Comment:
We use the empty string for tags with no values

Reply:
Is that a convention or do we count on it? I think we should support true/false as that is often more convenient when doing programmatic generation.

Reply:
We special-case the empty string to not show the value in the UI.

It was the result of this conversation: https://dagsterlabs.slack.com/archives/C04PD3G2F7H/p1711139737671339

Reply:
Open to revisiting. I think part of the consideration is supporting people who actually want the string
"some_tag_key=true" or "some_tag_key=false" to display in the UI.

Reply:
Got it. Let's keep the current convention then,